src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -467,6 +475,73 @@ bool CascadiaSettings::_isPowerShellCoreInstalledInPath(const std::wstring_view
     return false;
 }
 
+// Function Description:
+// - Adds all of the WSL profiles to the provided container.
+// Arguments:
+// - A ref to the profiles container where the WSL profiles are to be added
+// Return Value:
+// - <none>
+void CascadiaSettings::_AppendWslProfiles(std::vector<TerminalApp::Profile>& profileStorage)
+{
+    wil::unique_handle readPipe;
+    wil::unique_handle writePipe;
+    SECURITY_ATTRIBUTES sa{ sizeof(sa), nullptr, true };
+    THROW_IF_WIN32_BOOL_FALSE(CreatePipe(&readPipe, &writePipe, &sa, 0));
+    STARTUPINFO si{};
+    si.cb = sizeof(si);
+    si.dwFlags = STARTF_USESTDHANDLES;
+    si.hStdOutput = writePipe.get();
+    si.hStdError = writePipe.get();
+    wil::unique_process_information pi{};
+    std::wstring command = L"wsl.exe --list";
+
+    THROW_IF_WIN32_BOOL_FALSE(CreateProcessW(nullptr, const_cast<LPWSTR>(command.c_str()), nullptr, nullptr,
+                                                TRUE, CREATE_NO_WINDOW, nullptr, nullptr, &si, &pi));
+    switch (WaitForSingleObject(pi.hProcess, INFINITE))
+    {
+    case WAIT_OBJECT_0:
+        break;
+    case WAIT_ABANDONED:
+    case WAIT_TIMEOUT:
+        THROW_HR(ERROR_CHILD_NOT_COMPLETE);
+    case WAIT_FAILED:
+        THROW_LAST_ERROR();
+    default:
+        THROW_HR(ERROR_UNHANDLED_EXCEPTION);
+    }
+    DWORD exitCode;
+    if ((GetExitCodeProcess(pi.hProcess, &exitCode) == false) || (exitCode != 0))
+    {
+        THROW_HR(E_INVALIDARG);
+    }
+    DWORD bytesAvailable;
+    THROW_IF_WIN32_BOOL_FALSE(PeekNamedPipe(readPipe.get(), nullptr, NULL, nullptr, &bytesAvailable, nullptr));
+    FILE* hPipe = _wfdopen(_open_osfhandle((intptr_t)readPipe.get(), _O_WTEXT | _O_RDONLY), L"r");
+        //don't call fclose on hPipe because the readPipe handle is managed by wil and this will cause an error.

```

Comment:
Then let's not use the hPipe local var, just construct the std::wfstream with the _wfdopen.

Reply:
Fixed.